src/tsd/QueryRpc.java

```diff
@@ -115,14 +122,45 @@ private void handleQuery(final TSDB tsdb, final HttpQuery query) {
       throw new BadRequestException(HttpResponseStatus.BAD_REQUEST, 
           e.getMessage(), data_query.toString(), e);
     }
-    
+
     Query[] tsdbqueries;
     try {
       tsdbqueries = data_query.buildQueries(tsdb);
     } catch(NoSuchUniqueName ex) {
       throw new BadRequestException(ex);
     }
+
     final int nqueries = tsdbqueries.length;
+    if (query.method() == HttpMethod.DELETE) {
+      final ArrayList<Deferred<Object>> deferredsDelete =
+        new ArrayList<Deferred<Object>>(nqueries);
+      final byte[] table = tsdb.getConfig().getString("tsd.storage.hbase.data_table").getBytes();
+      final HBaseClient client = tsdb.getClient();
+
+      LOG.info("deleting from table=" + Arrays.toString(table));
+      try {
+        for (final Query q : tsdbqueries) {
+          q.setEndTime(Integer.MAX_VALUE);
+          final Scanner scanner = Internal.getScanner(q);
```

Comment:
So I should modify the `ScannerCB` class in both SaltScanner and TsdbQuery to delete the data if a flag was given, and then return an empty `Span` for each row? If so, I don't see how to integrate the `DeleteRequest` in this process. Any advice?